lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -760,7 +760,10 @@ def _on_config_changed(self, event: ConfigChangedEvent):  # noqa C901
             event.defer()
             return
 
-        if plugin_needs_restart or perf_profile_needs_restart:
+        if (
+            self.opensearch.is_service_started()
```

Comment:
Even if the `is_node_up()` returns false, the `service` may be up and, hence the JVM and its memory allocation already kicked in. Hence, we need to restart if we have the service up, not if we have the node up (which may happen later).